src/main/java/com/example/demo/WorkWithJsonFile.java

```diff
@@ -0,0 +1,65 @@
+package com.example.demo;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.reflect.TypeToken;
+import com.google.gson.stream.JsonReader;
+import org.springframework.stereotype.Repository;
+import java.io.*;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.List;
+
+
+@Repository
+public class WorkWithJsonFile {
```

Comment:
Please, consider reading a file during context initialization(remember bean lifecycle?). You can use lifecycle events on boot time to read file into collection and save collection on application shutdown